lib/navigation/control/navigator.cpp

```diff
@@ -127,4 +132,65 @@ core::Result Navigator::accelerometerUpdate(
   return core::Result::kSuccess;
 }
 
-}  // namespace hyped::navigation
+void Navigator::publishCurrentTrajectory()
+{
+  const auto topic     = core::MqttTopic::kTest;
+  auto message_payload = std::make_shared<rapidjson::Document>();
+  message_payload->SetObject();
+
+  auto trajectory = currentTrajectory();
+  if (!trajectory) { logger_.log(core::LogLevel::kFatal, "Failed to get current trajectory"); }
+
+  rapidjson::Value displacement(trajectory->displacement);
+  rapidjson::Value velocity(trajectory->velocity);
+  message_payload->AddMember("displacement", displacement, message_payload->GetAllocator());
+  message_payload->AddMember("velocity", velocity, message_payload->GetAllocator());
+
+  const core::MqttMessage::Header header{.timestamp = 0,
+                                         .priority  = core::MqttMessagePriority::kNormal};
+  const core::MqttMessage message{topic, header, message_payload};
+  mqtt_->publish(message, core::MqttMessageQos::kExactlyOnce);
+}
+
+void Navigator::run()
+{
+  mqtt_->subscribe(core::MqttTopic::kKeyence);
+  mqtt_->subscribe(core::MqttTopic::kOpticalFlow);
+  mqtt_->subscribe(core::MqttTopic::kAccelerometer);
+
+  while (true) {
+    mqtt_->consume();
+
+    auto keyence_msg       = mqtt_->getMessage();
+    auto optical_msg       = mqtt_->getMessage();
+    auto accelerometer_msg = mqtt_->getMessage();
+
+    auto keyence_payload          = keyence_msg->payload;
+    auto optical_payload          = optical_msg->payload;
+    auto raw_acceleration_payload = accelerometer_msg->payload;
+
+    core::KeyenceData dummy_keyence_data = {0, 0};
+
+    core::OpticalData dummy_optical_data;
+    for (auto &data : dummy_optical_data) {
+      data = {0.0F, 0.0F};
+    }
+
+    std::array<core::RawAccelerationData, core::kNumAccelerometers> dummy_accelerometer_data
+      = {core::RawAccelerationData(0, 0, 0, core::TimePoint{}, false),
+         core::RawAccelerationData(0, 0, 0, core::TimePoint{}, false),
+         core::RawAccelerationData(0, 0, 0, core::TimePoint{}, false),
+         core::RawAccelerationData(0, 0, 0, core::TimePoint{}, false)};
+
+    if (keyenceUpdate(dummy_keyence_data) == core::Result::kFailure
+        || opticalUpdate(dummy_optical_data) == core::Result::kFailure
+        || accelerometerUpdate(dummy_accelerometer_data) == core::Result::kFailure) {
```

Comment:
do `keyence_update_result = keyenceUpdate()` etc, then the if statement instead of blocking it all together